From bbb962065cb843dd18f4e90bfc3a793465ce1bbd Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Mon, 22 Feb 2010 16:53:27 +0100
Subject: [PATCH] net: mac80211: Make the softirq_count warning nonrt

commit c6dafd2110e83a8e34aefcbb7b1cb764f3e9d436 in tip.

ieee80211_rx() warns about softirq_count() == 0, which is always the
case on RT enabled kernels. Make the warning nonrt only.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
---
 net/mac80211/rx.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/mac80211/rx.c b/net/mac80211/rx.c
index 1946f6b..4d67bb8 100644
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -2494,7 +2494,7 @@ void ieee80211_rx(struct ieee80211_hw *hw, struct sk_buff *skb)
 	struct ieee80211_supported_band *sband;
 	struct ieee80211_rx_status *status = IEEE80211_SKB_RXCB(skb);
 
-	WARN_ON_ONCE(softirq_count() == 0);
+	WARN_ON_ONCE_NONRT(softirq_count() == 0);
 
 	if (WARN_ON(status->band < 0 ||
 		    status->band >= IEEE80211_NUM_BANDS))
-- 
1.7.0.4

